src/mapping/RadialBasisFctMapping.hpp

```diff
@@ -56,7 +67,12 @@ class RadialBasisFctMapping : public RadialBasisFctBaseMapping<RADIAL_BASIS_FUNC
 private:
   precice::logging::Logger _log{"mapping::RadialBasisFctMapping"};
 
+#ifndef PRECICE_NO_GINKGO
+  GinkgoRadialBasisFctSolver<RADIAL_BASIS_FUNCTION_T> _ginkgoRbfSolver;
+#endif
```

Comment:
I would probably prefer a `std::conditional` here and make the backend-type a template argument, such that we have.

```cpp
SolverBackend = std::conditional<GinkgoRadialBasisFctSolver, RadialBasisFctSolver, useEigen>;
```
Should remove all the if's in the code and we avoid the (large) branches for the compiler.

Reply:
I would highly prefer the solver to be a template argument to the RadialBasisFctMapping.

In fact, this means that the RadialBasisFunction isn't even required by the mapping itself any more, as it is purely responsible to shovel data around and call the RBFSolver.

```diff
-template <typename RADIAL_BASIS_FUNCTION_T>
+template <typename RBFSolver>
```

Then you would just create the RBFSolver here

```suggestion
  RBFSolver _solver;
```

Reply:
Sounds good, but I need to check how to do correctly.